File: script/capture-heartbeat.pl
#!/usr/bin/perl

use strict;
use warnings;
use 5.10.0;

use ZMQ::LibZMQ3;
use ZMQ::Constants qw( ZMQ_SUB ZMQ_SUBSCRIBE ); 

my $ctx = zmq_init();
die( "zmq_init failed" ) if( !$ctx );

my $socket = zmq_socket( $ctx, ZMQ_SUB );
die( "zmq_socket failed" ) if( !$socket );

my $rc = zmq_connect( $socket, $ARGV[ 0 ] );
die( "Connect failed" ) if( $rc != 0 );

$rc = zmq_setsockopt( $socket, ZMQ_SUBSCRIBE, "" ); 
die( "Set socket options failed" ) if( $rc != 0 );

my $msg = zmq_recvmsg( $socket, 0 );
my $str = zmq_msg_data( $msg );
say( $str );

$rc = zmq_msg_close( $msg );
$rc = zmq_close( $socket );
$rc = zmq_term( $ctx );

File: script/capture-heartbeat.rb
#!/usr/bin/env ruby
require 'bundler'
Bundler.setup
require 'ffi-rzmq'

begin
  ctx = ZMQ::Context.new
  socket = ctx.socket ZMQ::SUB
  puts "Connecting to locator at #{ARGV[0]}"
  rc = socket.connect( ARGV[0] )
  raise "Connect failed" if rc != 0
  rc = socket.setsockopt( ZMQ::SUBSCRIBE, "" ) 
  raise "Set socket options failed" if rc != 0
  message = ""
  count = 0
  while count < 3
    rc = socket.recv_string( message, ZMQ::DONTWAIT )
    break if rc >= 0
    sleep 1
    count += 1
  end
  #raise "Failed to receive expected messages error #{ZMQ::Util.error_string}" if rc < 0
  puts "Heartbeat => #{ message }"
  
rescue => e
  puts "script failed #{e.message}"
  puts e.backtrace.join( "\n" )
  exit 1
end

exit 0

